### PR TITLE
feat: discounts page new filter

### PR DESCRIPTION
**Pull request by DemianKryk**
## Type of Change

- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🛠️ **Bug** fix (non-breaking change which fixes an issue)
- [ ] 🧪 **Test** - Adds or updates tests.
- [ ] ❌ **Breaking** change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 **Code refactor**
- [ ] ✅ **Build configuration change**
- [ ] 📝 **Documentation**
- [ ] 🗑️ **Chore**

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -21,7 +21,7 @@ class App extends StatelessWidget {
       providers: [
         BlocProvider(
           create: (context) => GetIt.I.get<DiscountWatcherBloc>()
-            ..add(const DiscountWatcherEvent.started(isEnglish: false)),
+            ..add(const DiscountWatcherEvent.started()),
```
Comment (herman-the-worm) [inline lib/app.dart:24]: if we only instantiate events on class creation we can make this event instantiate privately within class constructor if you'd like

File: lib/components/discounts/bloc/advanced_filter/advanced_filter_mob_cubit.dart [hunk 2]
```diff
@@ -13,27 +12,57 @@ class AdvancedFilterMobCubit extends Cubit<AdvancedFilterMobState> {
       : super(
           const AdvancedFilterMobState(
             filtersLocation: [],
```
Comment (herman-the-worm) [inline lib/components/discounts/bloc/advanced_filter/advanced_filter_mob_cubit.dart:14]: chosen is a good word but I would prefer "activeLocationList" and similar across items 


Here are some chatgpt suggestions

For Active Data
selectedFilters: Clearly identifies the filters currently in use.
activeFilters: Indicates the filters actively being applied.
currentFilters: Suggests the filters currently selected by the user.
appliedFilters: Reflects the filters that are currently applied to the data.

**Review by herman-the-worm — APPROVED** [review]
Looks pretty good so far

File: lib/app.dart (hunk 1 above)
Comment (DemianKryk) [reply lib/app.dart:24]: I think I will create a separate task for this

Comment (DemianKryk) [reply lib/app.dart:24]: I think I will create a separate task for this

**Pull request merged**